src/ingestClient/ingestApiClient.ts

```diff
@@ -1,58 +1,39 @@
-import axios, { AxiosInstance } from 'axios';
-import axiosRetry from 'axios-retry';
-import * as Constants from '../model/constants';
+import BaseClient from "../baseClient";
 import { MeterMessage } from '../model/meterMessage';
 
-export class IngestApiClient {
-    axiosInstance: AxiosInstance;
-    signature: string;
+export class IngestApiClient extends BaseClient {
 
-    constructor(apiKey: string) {
-        this.signature = '[amberflo-metering IngestApiClient]:';
-        this.axiosInstance = axios.create({
-            baseURL: Constants.amberfloBaseUrl,
-            headers: {
-                "X-API-KEY": apiKey,
-                "Content-Type": "application/json"
-            },
-            timeout: 30000
-        });
-        axiosRetry(this.axiosInstance, {
-            retries: 3,
-            retryDelay: axiosRetry.exponentialDelay
-        });
+    constructor(apiKey: string, debug = false, retry = true) {
+        super(apiKey, debug, 'IngestApiClient', retry);
     }
 
     post(payload: Array<MeterMessage>, requestId: string, done?: () => void) {
-        // console.log(new Date(), this.signature, 'calling Ingest API with Request ID', requestId);
+        this.logDebug('about to post meters:', requestId);
         return this.axiosInstance
-            .post('/ingest-endpoint', payload)
+            .post('/ingest', payload)
             .then((response) => {
-                console.log(new Date(), this.signature, "response from Ingest API: ", requestId, response.status, response.data);
-                if (response.status >= 300) {
-                    console.log(`${this.signature} call to Ingest API failed ${response.status}, ${response.data}`);
-                }
+                this.logDebug('request completed:', requestId, response.status, response.data);
                 if (done) {
                     done();
                 }
             })
             .catch((error) => {
-                console.log(this.signature, new Date(), `call to Ingest API failed ${error}`);
+                this.logError('request failed:', requestId, error);
                 if (done) {
                     done();
                 }
             });
     }
 
     async postSync(payload: Array<MeterMessage>, requestId: string) {
-        // console.log(new Date(), this.signature, 'calling Ingest API with Request ID synchronously', requestId);
+        this.logDebug('about to post meters:', requestId);
         try {
             const response = await this.axiosInstance.post<string>('/ingest', payload);
             const data = await response.data;
-            console.log(new Date(), this.signature, 'request completed:', requestId, response.status, data);
+            this.logDebug('request completed:', requestId, response.status, data);
```

Comment:
we should check the api response status, and if it's not 200 then we should log it as an error

Reply:
log both the status and body

Reply:
Addressed